tests/helpers/datasets.py

```diff
@@ -212,3 +212,40 @@ def __getitem__(self, idx):
 
     def __len__(self):
         return len(self.y)
+
+
+class RandomDictDataset(Dataset):
```

Comment:
let upstream also these to the ain PL?

Reply:
I would rather do that in a separate PR. This one is already quite long. And I don't think all should be upstreamed.